content/articles/build-digital-clock-using-javascript/index.md

```diff
@@ -0,0 +1,309 @@
+
+What then is a ternary operator? The ternary operator is a conditional operator that takes three operands. The first being condition followed by a question mark (?). The second is an expression to execute if the condition is true, followed by a colon (:). Finally, the third operand is another expression to execute if the condition is false. Simple right? It is a simpler way to write an if-else statement.
+
+Here is the code snippet to illustrate that:
+
+```js
+hourOfDay = hourOfDay < 10 ? "0" + hourOfDay : hourOfDay;
+// if hourOfDay is less than 10, set hour position to 0 plus hourOfDay else set position to hourOfDay
+minutes = minutes < 10 ? "0" + minutes : minutes;
+// if minutes is less than 10, set minutes position to 0 plus minutes else set position to minutes
+seconds = seconds < 10 ? "0" + seconds : seconds;
+// if seconds is less than 10, set seconds position to 0 plus seconds else set position to seconds
+```
+
+I hope the ternary operator is clear to you now.
+
+We are almost done, the next thing to do is to set our current time. To do that, we need to declare a variable called ```currentTime```. Here is how:
+
+```js
+let currentTime = hourOfDay + ":" + minutes + ":" + seconds + period;
+```
+
+We initialize the ```currentTime``` variable with ```hourOfDay```, a colon, ```minutes```, a colon, ```seconds```, and ```period```. So for instance if the hour is 10, minutes is 12, seconds is 23, and period is AM, ```currentTime``` will be 10:12:23AM. This is clear, right? Two more things and we're done with the function body and JavaScript part of this article.
+
+Our next task is to display our clock on the web page and to do this, we need our JavaScript to manipulate our HTML. This is where the knowledge of the Document Object Model (DOM) becomes handy.
+
+There are various document methods but in this tutorial, we'll be using  ```document.getElementById```. With this, we'll be able to access an HTML element with a particular *id*. In this case, we will access the element with the id *Digital_Clock*.
+
+Here is the code snippet to illustrate how:
+
+```js
+document.getElementById('Digital_Clock').innerHTML =  currentTime + " " + today +" " + months + " " + year;
+```
+
+The ```innerHTML``` is a property of the DOM that sets or returns the HTML content of an element. In this case, the content of the div tag with an id of ```Digital_Clock``` is set to show the values of ```currentTime```, ```today```, ```months```, and ```year```. 
+
+The last thing is a function call or invocation. After our function has been declared, given a function body, it will still not work unless it is invoked or called. This is done by placing the function name and brackets immediately after the function's closing curly braces (}).
+
+This is how our final code will look:
+
+```js
+
+setInterval(displayTime, 1000);
+
+function displayTime() {
+    const hoursOfDay = timeNow.getHours();
```

Comment:
Missing the declaration for `timeNow` in this code snippet, and `hoursOfDay` set to be a constant but you are reassigning the value later on, this is not allowed.

I will try to fix this myself so don't worry about it.

Reply:
> Missing the declaration for `timeNow` in this code snippet, and `hoursOfDay` set to be a constant but you are reassigning the value later on, this is not allowed.
> 
> I will fix this myself so don't worry about it. I just wanted to let you know.

I already declared `timeNow` above in the code structure that's why I didn't bother to declare it here again as I was adding more function body. Thank you for fixing it.